src/keccak.rs

```diff
@@ -0,0 +1,262 @@
+use crunchy::unroll;
+use super::{PLEN, KeccakFamily, RHO, PI, Permutation};
+
+const RC: [u64; 24] = [
+    1u64,
+    0x8082u64,
+    0x800000000000808au64,
+    0x8000000080008000u64,
+    0x808bu64,
+    0x80000001u64,
+    0x8000000080008081u64,
+    0x8000000000008009u64,
+    0x8au64,
+    0x88u64,
+    0x80008009u64,
+    0x8000000au64,
+    0x8000808bu64,
+    0x800000000000008bu64,
+    0x8000000000008089u64,
+    0x8000000000008003u64,
+    0x8000000000008002u64,
+    0x8000000000000080u64,
+    0x800au64,
+    0x800000008000000au64,
+    0x8000000080008081u64,
+    0x8000000000008080u64,
+    0x80000001u64,
+    0x8000000080008008u64,
+];
+
+/// keccak-f[1600]
+#[allow(unused_assignments)]
+pub fn keccakf(a: &mut [u64; PLEN]) {
```

Comment:
Unless I'm mistaken it looks exactly like the function in `KangarooTwelve` but with a different `RC`. Can't we extract this to a common function or a macro?

Also same grumbles apply here.